File: C2203L/JP1/07-11-2022/abstractExample/src/main/java/com/ab5tract/example/Car.java
package com.ab5tract.example;

public class Car extends Vehicle{
    private int numberOfDoors;

    public Car(String brand, String model, String registrationNumber, Person owner,
               int numberOfDoors) {
        super(brand, model, registrationNumber, owner);
        this.numberOfDoors = numberOfDoors;
    }

    @Override
    public String toString() {
        return super.toString() + "numberOfDoors:"+numberOfDoors;
    }
}

File: C2209GDoiCan/IDP/13-12-2024/car.dart
import 'vehicle.dart';

class Car extends Vehicle {
  int? numberOfDoors;
  Car({required String brand, required int speed, required this.numberOfDoors})
      : super(brand: brand, speed: speed);
  @override
  void displayInfo() {
    print(
        'brand: ${super.brand}, speed: ${super.speed}, numberOfDoors: ${numberOfDoors}');
  }
}
